netket/utils/array.py

```diff
@@ -32,14 +31,13 @@ class HashableArray:
     wrapped: Array
     """The wrapped array. Note that this array is read-only."""
 
-    def __post_init__(self):
-        object.__setattr__(self, "wrapped", self.wrapped.copy())
-        if isinstance(self.wrapped, np.ndarray):
-            self.wrapped.flags.writeable = False
-        object.__setattr__(self, "_HashableArray__hash", hash(self.wrapped.tobytes()))
+    def __pre_init__(self, wrapped):
+        wrapped = wrapped.copy()
+        wrapped.flags.writeable = False
+        return (wrapped,), {}
```

Comment:
This will fail if `wrapped` is not an `ndarray` but a jax `DeviceArray`. (Because that one does not provide the `writeable` flag.) The `isinstance(self.wrapped, np.ndarray)` check needs to be added back.